Automattic/wp-calypso
# SDK: Rely on preset-provided editor.js

#### Changes proposed in this Pull Request

Follow-up to #29435, to remove more duplication, and make the SDK's treatment of presets align more closely to that of blocks:

- Move `getExtensions()` to `utils/get-extensions.js`
- Change the Jetpack preset to register all blocks from a newly introduced `editor.js`
- Drop editor script logic from `bin/sdk/gutenberg.js`, rely on that new `editor.js` as the entrypoint (also allows removing a bunch of beta-blocks related logic)

#### TODO

Figure out module-level registrations etc in shared/util dirs.

#### Testing instructions

- Test in Calypso, verify that it still works
- Test in JP, verify that it still works
- Pay special attention to the Jetpack sidebar (that contains the Publicize extension), since that was previously directly included.
